pkg/autodiscover/autodiscover.go

```diff
@@ -280,3 +290,81 @@ func getOpenshiftVersion(oClient clientconfigv1.ConfigV1Interface) (ver string,
 
 	return "", errors.New("could not get openshift version from clusterOperator")
 }
+
+// Get a map of csvs with its managed pods from the target namespaces
+func getOperatorCsvPods(csvList []*olmv1Alpha.ClusterServiceVersion) (map[string][]*corev1.Pod, error) {
+	client := clientsholder.GetClientsHolder()
+	csvToPodsMapping := make(map[string][]*corev1.Pod)
+
+	for _, csv := range csvList {
+		// Get target namespaces
+		operatorTargetNamespaces, err := getOperatorTargetNamespaces(csv, client)
+		if err != nil {
+			return csvToPodsMapping, err
+		}
+		var podList []*corev1.Pod
+
+		for _, targetNamespace := range operatorTargetNamespaces {
+			pods, err := getCsvPodsFromTargetNamespace(csv, strings.TrimSpace(targetNamespace), client)
+			if err != nil {
+				continue
+			}
+			podList = append(podList, pods...)
+		}
+
+		csvToPodsMapping[fmt.Sprintf("%s, namespace=%s", csv.Name, csv.Namespace)] = podList
```

Comment:
create a constant with "%s, ns=%s", since the user of this map will need to use it too

Reply:
Sure, but we do not need this as @sebrandon1 suggested to add csv info in provider.Pod.